rmpv/src/decode/mod.rs

```diff
@@ -17,6 +20,16 @@ pub enum Error {
     InvalidMarkerRead(io::Error),
     /// Error while reading data.
     InvalidDataRead(io::Error),
+    /// The depth limit [`MAX_DEPTH`] was exceeded.
+    DepthLimitExceeded,
```

Comment:
Sadly, this error is public and not marked as non-exhaustive, so new variants are semver-major breaking. 
Do you want to break semver for this, or maybe make an `Other` kind of io::Error?

Reply:
I'd like to be consistent with `rmpv-serde`, which has the same dedicated error. Sounds like possibly bumping `rmpv` to 0.5, then?